### PR TITLE
feat(ingest/kafka-connect): Lineage for Kafka Connect > Snowflake

### PR DESCRIPTION
**Pull request by shubhamjagtap639**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by mayurinehate — APPROVED** [review]
lgtm

File: metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py [hunk 1]
```diff
@@ -879,6 +879,106 @@ def _extract_lineages(self):
...
+
+    def __init__(
+        self, connector_manifest: ConnectorManifest, report: KafkaConnectSourceReport
+    ) -> None:
+        self.connector_manifest = connector_manifest
+        self.report = report
+        self._extract_lineages()
+
+    @dataclass
+    class SnowflakeParser:
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py:895]: nested classes are generally an anti-pattern. is there a reason you're using them here?

File: metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py [hunk 2]
```diff
@@ -879,6 +879,106 @@ def _extract_lineages(self):
...
+        )
+
+    def _extract_lineages(self):
+        self.connector_manifest.flow_property_bag = self.connector_manifest.config
+
+        # remove private keys, secrets from properties
+        secret_properties = [
+            "snowflake.private.key",
+            "snowflake.private.key.passphrase",
+            "value.converter.basic.auth.user.info",
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py:958]: are these the only private properties?

**Review by hsheth2 — COMMENTED** [review]
small nits/questions but otherwise looking good

**Review by mayurinehate — COMMENTED** [review]
@shubhamjagtap639 can you please update limitations section in kafka connect documention to reflect all sink connectors supported. I believe confluent s3 sink connector and snowflake sink connector are missing there.

File: metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py (hunk 2 above)
Comment (shubhamjagtap639) [reply metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py:958]: Yes we checked all properties and this are the only private properties that we found at https://docs.snowflake.com/en/user-guide/kafka-connector-install#configuring-the-kafka-connector

Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py:958]: cool could you add that link as a comment in the code

File: metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py:895]: ok that's fien

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**